src/pull_module/gguf_downloader.cpp

```diff
@@ -0,0 +1,249 @@
+//*****************************************************************************
+// Copyright 2025 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include "gguf_downloader.hpp"
+
+#include <string>
+#include <memory>
+#include <iostream>
+
+#include <curl/curl.h>
+#include <nlohmann/json.hpp>
+
+#include "../capi_frontend/server_settings.hpp"
+#include "../filesystem.hpp"
+#include "../localfilesystem.hpp"
+#include "../logging.hpp"
+#include "../stringutils.hpp"
+#include "../status.hpp"
+#include "../version.hpp"
+
+namespace ovms {
+std::string GGUFDownloader::getGraphDirectory() {
+    return this->downloadPath;
+}
+static Status checkIfOverwriteAndRemove(const HFSettingsImpl& hfSettings, const std::string& path) {
+    auto lfstatus = StatusCode::OK;
+    if (hfSettings.overwriteModels && std::filesystem::is_directory(path)) {
+        LocalFileSystem lfs;
+        lfstatus = lfs.deleteFileFolder(path);
```

Comment:
any locking required here?

Reply:
The class  functionality does not depend on existing file, this is just pin pointed removal if user wants to overwrite.